### DISCUSSION TITLE
type must be array but is object [nlohmann::details::type_error]

### DISCUSSION THREAD

**Discussion by hrishi77** [nlohmann/json#2497] [Q&A] [answered]
```cpp
// inherit class 
namespace handler{
    class json: public nlohmann::json {
     // functions
    };  
}

//This one one works fine  but below code gives error ( exception ) 
nlohmann::json parseConf(const tstring &s)
 {
   nlohmann::json configJSON = nlohmann::json::parse(s);
   return configJSON;
 }

// error 
handler::json parseConf(const tstring &s)
 {
   return handler::json::parse(s);
 }
```

type must be array but is object

**Comment by nlohmann** [comment, marked as answer]
Can you please provide more information?

Also note the `nlohmann::json` is not really designed to be inherited from.

**Reply by hrishi77** [reply]
inherited nlohmann::json as base class ,i am able to access methods/functions (like parse, at,[]operator ) from  nlohmann::json in derived class.
In my above first example i am taking return type as nlohmann::json but when use derived::json (my own class) that time it gives error type must be array not object .I did not understand why ? derived class type should behave like base but it's not working

**Reply by nlohmann** [reply]
What is the exact exception output? There should be an id. Do you have a stack trace?

**Reply by hrishi77** [reply]
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.302] type must be array, but is object
Aborted

**Reply by nlohmann** [reply]
Stacktrace?

**Reply by hrishi77** [reply]
not able to give
